common/lib/xmodule/xmodule/js/src/sequence/display.js

```diff
@@ -0,0 +1,354 @@
+            };
+
+            this.updatedProblems = {};
+            this.requestToken = $(element).data('request-token');
+            this.el = $(element).find('.sequence');
+            this.path = $('.path');
+            this.contents = this.$('.seq_contents');
+            this.content_container = this.$('#seq_content');
+            this.sr_container = this.$('.sr-is-focusable');
+            this.num_contents = this.contents.length;
+            this.id = this.el.data('id');
+            this.ajaxUrl = this.el.data('ajax-url');
+            this.nextUrl = this.el.data('next-url');
+            this.prevUrl = this.el.data('prev-url');
+            this.base_page_title = ' | ' + document.title;
+            this.bind();
+            this.render(parseInt(this.el.data('position'), 10));
+        }
+
+        Sequence.prototype.$ = function(selector) {
+            return $(selector, this.el);
+        };
+
+        Sequence.prototype.bind = function() {
+            this.$('#sequence-list .nav-item').click(this.goto);
+            this.el.on('bookmark:add', this.addBookmarkIconToActiveNavItem);
+            this.el.on('bookmark:remove', this.removeBookmarkIconFromActiveNavItem);
+            this.$('#sequence-list .nav-item').on('focus mouseenter', this.displayTabTooltip);
+            this.$('#sequence-list .nav-item').on('blur mouseleave', this.hideTabTooltip);
+        };
+
+        Sequence.prototype.displayTabTooltip = function(event) {
+            $(event.currentTarget).find('.sequence-tooltip').removeClass('sr');
+        };
+
+        Sequence.prototype.hideTabTooltip = function(event) {
+            $(event.currentTarget).find('.sequence-tooltip').addClass('sr');
+        };
+
+        Sequence.prototype.updatePageTitle = function() {
+            // update the page title to include the current section
+            var positionLink = this.link_for(this.position);
+
+            if (positionLink && positionLink.data('page-title')) {
+                document.title = positionLink.data('page-title') + this.base_page_title;
+            }
+        };
+
+        Sequence.prototype.hookUpContentStateChangeEvent = function() {
+            var self = this;
+
+            return $('.problems-wrapper').bind('contentChanged', function(event, problemId, newContentState, newState) {
+                return self.addToUpdatedProblems(problemId, newContentState, newState);
+            });
+        };
+
+        Sequence.prototype.addToUpdatedProblems = function(problemId, newContentState, newState) {
+            /**
+            * Used to keep updated problem's state temporarily.
+            * params:
+            *   'problem_id' is problem id.
+            *   'new_content_state' is the updated content of the problem.
+            *   'new_state' is the updated state of the problem.
+            * initialize for the current sequence if there isn't any updated problem
```

Comment:
This comment (line 103) belongs to the line of code at 107, instead of top-level documentation for the method.

That being said, I'm not sure the comment is particularly useful (the method name is self-documenting, which is good).